### PR TITLE
No bids specific nifti files in the dataset

### LINKED ISSUE #325: Raise an error if there are no subject specific BIDS compatible files
Here's a dataset that technically passes the validator:
image

But the dataset is defunct due to typos:
image

It would be useful that in such cases (where there are no BIDS compatible files that are specific to any given participant) the validator returned an error.

### PR DESCRIPTION
**Pull request by suyashdb**
Closes #325

### CODE REVIEW

File: validators/bids.js [hunk 1]
```diff
@@ -232,6 +241,7 @@ BIDS = {
             // capture niftis for later validation
             else if (file.name.endsWith('.nii') || file.name.endsWith('.nii.gz')) {
                 niftis.push(file);
+                nifti_fileList.push(file.relativePath);
```
Comment (chrisgorgo) [inline validators/bids.js]: @suyashdb : "This has to be removed"

File: validators/bids.js [hunk 2]
```diff
@@ -372,6 +383,8 @@ BIDS = {
 
         }, function () {
             async.eachOfLimit(niftis, 200, function (file, key, cb) {
+                
+                nifti_fileList.push(file.relativePath);
```
Comment (chrisgorgo) [inline validators/bids.js]: Why is this added again?

**Comment by chrisgorgo** [comment]
Please iterate over subjects and give error for every subject that doesn't have any BIDS specific files.

File: validators/bids.js [hunk 3]
```diff
@@ -199,6 +200,14 @@ BIDS = {
             }
         });
 
+        // collecting all nifti files into an array
+        async.eachOfLimit(fileList, 200, function (file, key, cb) {
+            if (file.name.endsWith('.nii') || file.name.endsWith('.nii.gz')) {
```
Comment (chrisgorgo) [inline validators/bids.js]: please don't restrict this test to niftis

**Review by chrisgorgo — CHANGES_REQUESTED** [review]

**Comment by chrisgorgo** [comment]
There was another dataset like this in OpenNeuro. What is the ETA on this PR?

**Comment by chrisgorgo** [comment]
I just realized that this test could be achieved by comparing `summary.subjects` count with number of `sub-*` subfolders. Should be very easy to implement.

**Comment by suyashdb** [comment]
I am aiming to submit it my today evening..

**Comment by suyashdb** [comment]
@chrisfilo:
I tried reducing the codeclimate complexity from 15 to 6. Not sure how can i reduce to 5 to pass codeclimate threshold. :)

CircleCI: as per my initial digging, linter is giving error for ES6 syntax. Trying on different linter if that works. 

Also, trying your suggestion " comparing summary.subjects count with number of sub-* subfolders. Should be very easy to implement."

**Comment by chrisgorgo** [comment]
https://github.com/ternjs/acorn/issues/500

**Pull request closed without merging**